src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -68,19 +69,36 @@ class Proxy(ORMMixin):
     def key_fields(cls) -> Tuple[str, Optional[str]]:
         return ("region", "proxy_id")
 
-    def get_vm(self) -> VM:
+    def get_vm(self) -> Union[VM, Error]:
+        nsg = NSG(
+            name=self.region,
+            region=self.region,
+        )
+
+        result = nsg.create()
```

Comment:
`get_vm` doesn't actually create VMs or any of the resources behind a VM at this time.    This doesn't behave like any of the other underlying resources associated with a VM, like nics, IPs, etc.